### PR TITLE
Add .clang-tidy and fix a few issues it detects

### PR DESCRIPTION
**Pull request by lanza**
clang-tidy detects a few minor issues. Fix them here.

### CODE REVIEW

File: Sources/Utils/POSIX/Stringify.cpp [hunk 1]
```diff
@@ -17,8 +17,8 @@
 #include <cerrno>
 #include <csignal>
 // clang-format off
-#include <sys/types.h>
-#include <sys/ptrace.h>
+#include <sys/types.h> // NOLINT
```
Comment (sas) [inline Sources/Utils/POSIX/Stringify.cpp]: Why is this required?

Comment (lanza) [reply Sources/Utils/POSIX/Stringify.cpp]: I don't know the fundamental reason why, but you added the `clang-format off` in 2016 because types needed to be included before ptrace.

Comment (lanza) [reply Sources/Utils/POSIX/Stringify.cpp]: That worked.
